Ewhair

comma opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

Headgear that blocks hair now makes it appear as 'Short Hair' instead of sudden baldness.
Unless you're bald of a xeno, then you get old behaviour yes.
Also fedoras now do that hair thing by default, click in hand to toggle old behaviour.

Raptor1628 commented:
How does this get along with spacesuit helmets? And what does this do for people with a hair sprite that's shorter than "short hair" like all the buzzcut and crewcut folks?

comma commented:
Wouldn't notice difference with helmets, since it's all hidden anyway.
As for all those damn hairstyles, I guess can't get away with simple list anymore.
Halway done with a fixie.

GinjaNinja32 commented on code/modules/clothing/under/jobs/security.dm:
```diff
@@ -104,10 +104,14 @@
+	flags_inv = BLOCKHEADHAIR
+
+/obj/item/clothing/head/det/attack_self(mob/user)
+	flags_inv = flags_inv & BLOCKHEADHAIR ? 0 : BLOCKHEADHAIR
```
`flags_inv ^= BLOCKHEADHAIR`?

GinjaNinja32 commented on code/modules/clothing/under/jobs/security.dm:
```diff
@@ -104,10 +104,14 @@
+
+/obj/item/clothing/head/det/attack_self(mob/user)
+	flags_inv = flags_inv & BLOCKHEADHAIR ? 0 : BLOCKHEADHAIR
+	user << "<span class='notice'>[src] will now [flags_inv ? "hide" : "show"] hair.</span>"
```
`flags_inv & BLOCKHEADHAIR`?

comma replied:
Detective hat only has that one flag, so it's equivalent.

GinjaNinja32 replied:
Currently, yes; what if another flag is added? This will be a hidden bug until someone notices it.

comma replied:
ugggh YAGNI, but fine, whatever, I just want this PR to be gone alreade, I'll add it.

PsiOmegaDelta reviewed (approved):
Could use a squash but other than that

comma commented:
See commit description

comma committed:
Hair stuff

Headgear that blocks hair now makes it appear as 'Short Hair' instead of sudden baldness.
Unless you're bald of a xeno, then you get old behaviour yes.
Also fedoras now do that hair thing by default, click in hand to toggle old behaviour.

comma commented:
okay, all unfucked
